src/vdom/diff.js

```diff
@@ -262,11 +262,12 @@ export function recollectNodeTree(node, unmountOnly) {
 		// (this is part of the React spec, and smart for unsetting references)
 		if (node[ATTR_KEY]!=null && node[ATTR_KEY].ref) node[ATTR_KEY].ref(null);
 
+		removeChildren(node);
```

Comment:
Then I would suggest to either call only the unmountComponent depth first, or wait until the most upper dirty node, to remove it.